src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +253,218 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
+                    FeatureManagementError.MissingFeatureVariant,
+                    $"No variants are registered for the feature {feature}");
+            }
+
+            FeatureVariant featureVariant;
+
+            bool isFeatureEnabled = await IsEnabledAsync(feature, context, true).ConfigureAwait(false);
+
+            featureVariant = await GetFeatureVariantAsync(featureDefinition, context, useContext, isFeatureEnabled, cancellationToken).ConfigureAwait(false);
+
+            if (featureVariant == null)
+            {
+                return null;
+            }
+
+            IConfigurationSection variantConfiguration = null;
+
+            bool configValueSet = !string.IsNullOrEmpty(featureVariant.ConfigurationValue);
+            bool configReferenceValueSet = !string.IsNullOrEmpty(featureVariant.ConfigurationReference);
+
+            if (configValueSet && configReferenceValueSet)
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidVariantConfiguration,
+                    $"Both ConfigurationValue and ConfigurationReference are specified for the variant {featureVariant.Name} in feature {feature}");
+            }
+            else if (configReferenceValueSet)
+            {
+                if (_configuration == null)
+                {
+                    throw new FeatureManagementException(
+                        FeatureManagementError.InvalidVariantConfiguration,
+                        $"Cannot use {nameof(featureVariant.ConfigurationReference)} if no instance of {nameof(IConfiguration)} is present.");
+                }
+
+                variantConfiguration = _configuration.GetSection(featureVariant.ConfigurationReference);
+            }
+            else if (configValueSet)
+            {
+                VariantConfigurationSection section = new VariantConfigurationSection(featureVariant.Name, "", featureVariant.ConfigurationValue);
+                variantConfiguration = section;
+            }
+
+            Variant returnVariant = new Variant()
+            {
+                Name = featureVariant.Name,
+                Configuration = variantConfiguration
+            };
+
+            return returnVariant;
+        }
+
+        private async ValueTask<FeatureVariant> GetFeatureVariantAsync(FeatureDefinition featureDefinition, TargetingContext context, bool useContext, bool isFeatureEnabled, CancellationToken cancellationToken)
+        {
+            if (!isFeatureEnabled)
+            {
+                return ResolveDefaultFeatureVariant(featureDefinition, isFeatureEnabled);
+            }
+
+            FeatureVariant featureVariant = null;
+
+            if (!useContext) 
+            {
+                if (_contextAccessor == null)
+                {
+                    _logger.LogWarning($"No instance of {nameof(ITargetingContextAccessor)} is available for targeting evaluation. Using default variants.");
```

Comment:
Yea makes sense, I guess the way it is now that isn't a possible scenario but I'd be open to changing this section since I'm not super sure how to handle this scenario. I wanted to do it this way so that even if we can't get `TargetingContext` the user could see a log for it, but I hope it isn't too many warning level logs.